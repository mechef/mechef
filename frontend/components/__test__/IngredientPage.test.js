import renderer from 'react-test-renderer';
import React from 'react';

import { IngredientPage } from '../IngredientPage';

describe('Pages With Snapshot Testing', () => {
  it('snapshot IngredientPage Component', () => {
    const component = renderer.create(
      <IngredientPage
        ingredient={{
          memos: [
            {
              _id: '59d054c8a70e5b90e75111b8',
              sum: '100',
              name: 'bible memo',
              ingredients: [
                {
                  name: 'ingredient2',
                  amount: '1234',
                },
              ],
            },
          ],
          currentMemoId: '59d054c8a70e5b90e75111b8',
        }}
        fetchMemos$={() => {}}
        createMemo$={() => {}}
        updateMemo$={() => {}}
        deleteMemo$={() => {}}
        setCurrentMemoId$={() => {}}
        setError$={() => {}}
        toggleBackArrow$={() => {}}
        error={{
          title: '',
          message: '',
          isShowModal: false,
        }}
        global={{
          backArrow: {
            isShow: false,
            title: '',
          },
        }}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
