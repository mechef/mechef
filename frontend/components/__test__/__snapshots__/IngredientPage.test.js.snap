// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot IngredientPage Component 1`] = `
<div
  className="jsx-2726819105 container"
>
  <div
    className="jsx-158552079"
  >
    <div
      className="jsx-158552079 header"
    >
      <span
        className="jsx-158552079 title"
      >
        Ingredients List
      </span>
      <div
        className="jsx-158552079 addButton"
        onClick={[Function]}
        role="link"
        tabIndex="-1"
      >
        <i
          aria-hidden="true"
          className="jsx-158552079 fa fa-plus plus-icon"
        />
      </div>
    </div>
    <div
      className="jsx-158552079 ingredient-list"
    >
      <div
        className="jsx-158552079 ingredient-item"
      >
        <div
          className="jsx-158552079 ingredient-content"
        >
          <p
            className="jsx-158552079 ingredient-title"
          >
            bible memo
          </p>
          <p
            className="jsx-158552079 ingredient-detail"
          >
            <span
              className="jsx-158552079 ingredient-subtext"
            >
              Ingredient: 
              1
            </span>
            <span
              className="jsx-158552079 ingredient-subtext"
            >
              Total: 
              $100
            </span>
          </p>
        </div>
        <span
          className="jsx-158552079 update-button"
          onClick={[Function]}
          role="button"
          tabIndex="-1"
        >
          <span
            className="jsx-158552079 update-button-text"
          >
            UPDATE
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
